### PR TITLE
mgmt core, move ArmChallengeAuthenticationPolicy from experimental

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
`ArmChallengeAuthenticationPolicy` moved and simplified from https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/policy/ArmChallengeAuthenticationPolicy.java

Simplify it with static `scopes` for simple Lite SDK (which only have mgmt-plane URL, hence no need to switch scope dynamically based on request URL).
Multiple `scopes` based on request URL might cause problem to the `AccessTokenCache cache` (a request to data-plane after request to mgmt-plane will force the cache update and discard the previous token for mgmt-plane). Ideally we might want to have different cache for different set of scopes.

Premium SDK would rely on `AccessTokenCache cache` for switch across different scopes.

### CODE REVIEW

**Comment by yungezz** [comment]
cc @msyyc

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/http/policy/ArmChallengeAuthenticationPolicy.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    }
+                }
+                return Mono.just(false);
+            }
+        });
+    }
+
+    protected String[] getScopes(HttpPipelineCallContext context, String[] scopes) {
+        return scopes;
+    }
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-management/src/main/java/com/azure/core/management/http/policy/ArmChallengeAuthenticationPolicy.java:103]: In azure-core-management, only use a simple static `scopes`.

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/AuthenticationPolicy.java [hunk 2]
```diff
@@ -36,32 +25,16 @@
...
-        if (this.scopes == null || this.scopes.length == 0) {
-            String defaultScope = ResourceManagerUtils.getDefaultScopeFromRequest(
-                context.getHttpRequest(), environment);
-            tokenResult = this.credential.getToken(new TokenRequestContext().addScopes(defaultScope));
-        } else {
-            tokenResult = this.credential.getToken(new TokenRequestContext().addScopes(scopes));
+    protected String[] getScopes(HttpPipelineCallContext context, String[] scopes) {
+        if (CoreUtils.isNullOrEmpty(scopes)) {
+            scopes = new String[1];
+            scopes[0] = ResourceManagerUtils.getDefaultScopeFromRequest(context.getHttpRequest(), environment);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/AuthenticationPolicy.java:36]: Dynamic `scopes` in azure-resourcemanager-resources.

It depends on `AccessTokenCache` to retrieve new token when scopes changed.

**Comment by weidongxu-microsoft** [comment]
msdocs https://docs.microsoft.com/en-us/azure/active-directory/develop/claims-challenge

**Comment by weidongxu-microsoft** [comment]
live test pass on Azure cloud

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/AuthenticationPolicy.java (hunk 2 above)
Comment (g2vinay) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/AuthenticationPolicy.java:36]: Yeah the AccessTokenCache will retreive a new token if a different scope is passed in.

**Review by g2vinay — APPROVED** [review]

**Pull request merged**